server/src/main/java/org/opensearch/common/logging/DeprecatedMessage.java

```diff
@@ -77,6 +82,34 @@ private static Map<String, Object> fieldMap(String key, String xOpaqueId) {
     }
 
     public boolean isAlreadyLogged() {
```

Comment:
The method is called from [DeprecationLoggerBuilder](https://github.com/rajiv-kv/OpenSearch/blob/0d54c1600b16081c15902a095111a9a530767099/server/src/main/java/org/opensearch/common/logging/DeprecationLogger.java#L120) and we end up tracking the keys even if logging in not enabled.

Should we add a check if DEPRECATION log is enabled before calling this method ?

Reply:
Yeah, that seems like a good idea